From: =?UTF-8?q?Alfonso=20S=C3=A1nchez-Beato?= <alfonso.sanchez-beato@canonical.com>
Date: Thu, 18 Jul 2019 17:48:39 +0200
Subject: [PATCH] snap: pass NM env vars when calling iptables

This is necessary so LD_LIBRARY_PATH is passed down to iptables so it is
able to find the libraries included in the snap package.
---
 src/nm-act-request.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/nm-act-request.c b/src/nm-act-request.c
index 87070794..bcf3e0bb 100644
--- a/src/nm-act-request.c
+++ b/src/nm-act-request.c
@@ -309,7 +309,6 @@ nm_act_request_set_shared (NMActRequest *req, gboolean shared)
 	/* Send the rules to iptables */
 	for (iter = list; iter; iter = g_slist_next (iter)) {
 		ShareRule *rule = (ShareRule *) iter->data;
-		char *envp[1] = { NULL };
 		gs_strfreev char **argv = NULL;
 		gs_free char *cmd = NULL;
 
@@ -327,7 +326,7 @@ nm_act_request_set_shared (NMActRequest *req, gboolean shared)
 			GError *error = NULL;
 
 			nm_log_info (LOGD_SHARING, "Executing: %s", cmd);
-			if (!g_spawn_sync ("/", argv, envp, G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
+			if (!g_spawn_sync ("/", argv, NULL, G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
 			                   NULL, NULL, NULL, NULL, &status, &error)) {
 				nm_log_warn (LOGD_SHARING, "Error executing command: %s",
 				             error->message);
-- 
2.17.1

